certbot/certbot/_internal/log.py

```diff
@@ -279,7 +279,8 @@ def close(self):
         try:
             # StreamHandler.close() doesn't close the stream to allow a
             # stream like stderr to be used
-            self.stream.close()
+            if hasattr(self.stream, 'close'):
+                self.stream.close()  # type: ignore
```

Comment:
If we're using `type: ignore` do we need the conditional? I personally just don't like branches like this because if `self.stream.close` doesn't exist, something has gone horribly wrong and I'd like us to blow up.
```suggestion
            self.stream.close()  # type: ignore
```
Other than this small comment, this LGTM!

Reply:
In fact I was not very satisfied with this ignore directive. Looking at the code of `logging.StreamHandler`, is see that when no `stream` object is provided to the super constructor, `sys.stderr` will be assigned to it, which is of type `typing.TypeIO` from mypy perspective.

Given that we effectively do not assign a specific `stream` here, I add the type information in the constructor to assess the effective type in `TempHandler`.

This also solves the mypy error in a more elegant manner I believe.

Reply:
In fact the super constructor of `TempHandler` will assign the provided `stream` object to the `self.stream` attribute. I added a type hint to the `TempHandler` constructor to let mypy be aware of this.

This solves also the issue, in a more elegant manner I believe.

Reply:
Fantastic. I agree your new approach is even better.